pandas/core/indexes/base.py

```diff
@@ -5930,10 +5917,11 @@ def trim_front(strings: List[str]) -> List[str]:
     """
     Trims zeros and decimal points.
     """
-    trimmed = strings
-    while len(strings) > 0 and all(x[0] == " " for x in trimmed):
-        trimmed = [x[1:] for x in trimmed]
-    return trimmed
+    if not strings:
+        return strings
+    while all(x[0] == " " for x in strings):
```

Comment:
is this going to break when one of the strings becomes empty?

Reply:
Yes, you're absolutely right, thanks! It fails on `master` too, but still, worth fixing while we're modifying these lines
```python
In [1]: from pandas.core.indexes.base import trim_front

In [2]: trim_front([' ', ' a'])
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-2-1d23255f182c> in <module>
----> 1 trim_front([' ', ' a'])

~/pandas-dev/pandas/core/indexes/base.py in trim_front(strings)
   5848     """
   5849     trimmed = strings
-> 5850     while len(strings) > 0 and all(x[0] == " " for x in trimmed):
   5851         trimmed = [x[1:] for x in trimmed]
   5852     return trimmed

~/pandas-dev/pandas/core/indexes/base.py in <genexpr>(.0)
   5848     """
   5849     trimmed = strings
-> 5850     while len(strings) > 0 and all(x[0] == " " for x in trimmed):
   5851         trimmed = [x[1:] for x in trimmed]
   5852     return trimmed

IndexError: string index out of range
```

Reply:
Looking at this again, I don't think it's an issue, because `trim_front` is only ever called with a list of strings which are all of the same length.

It's only ever called from `pd.Index._format_with_header`:
```
result = trim_front(format_array(values, None, justify="left"))
```
and `format_array` from `pandas/io/formats/format.py` returns
```
fmt_obj.get_result()
```
which in turn returns
```
_make_fixed_width(fmt_values, self.justify)
```

_Nonetheless_ I can make the condition
```
    while all(strings) and all(x[0] == " " for x in strings):
```
and add a tiny test for which that'd be necessary